File: SNACKS/Models/Persona.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SNACKS.Models
{
    public class Persona
    {
        public int IdPersona { get; set; }
        public int TipoPersona { get; set; }
        public string Nombres { get; set; }
        public string Apellidos { get; set; }
        public string RazonSocial { get; set; }
        public int TipoDocumento { get; set; }
        public string NumeroDocumento { get; set; }
        public string Distrito { get; set; }
        public string Direccion { get; set; }
        public ZonaVenta ZonaVenta { get; set; }
        public Persona Vendedor { get; set; }
    }
}

File: SNACKS/ClientApp/src/app/personas/persona.ts
import { IZonaVenta } from "../zonas-venta/zonaVenta";

export interface IPersona {
  idPersona: number;
  tipoPersona: number;
  nombres: string;
  apellidos: string;
  razonSocial: string;
  tipoDocumento: number;
  numeroDocumento: string;
  direccion: string;
  distrito: string;
  zonaVenta: IZonaVenta;
  vendedor: IPersona;
}
